File: client/src/domain/ShotResult.ts
import { Ship } from "./index";
import { GameState } from "utils/Constants";

export default class ShotResult {
  readonly grid: number[][];
  readonly ship?: Ship;
  readonly gameState: GameState;

  constructor(grid: number[][], gameState: GameState, ship?: Ship) {
    this.grid = grid;
    this.ship = ship;
    this.gameState = gameState;
  }
}

File: server/battleships/domain/src/main/java/eu/optas/domain/ShotResult.java
package eu.optas.domain;

import eu.optas.utils.GameState;

import java.util.List;
import java.util.Optional;

public class ShotResult {
    private final List<List<Integer>> grid;
    private final Ship ship;
    private final GameState gameState;

    public ShotResult(List<List<Integer>> grid, Ship ship, GameState gameState) {
        this.grid = grid;
        this.ship = ship;
        this.gameState = gameState;
    }

    public List<List<Integer>> getGrid() {
        return grid;
    }

    public Optional<Ship> getShip() {
        return Optional.ofNullable(ship);
    }

    public GameState getGameState() {
        return gameState;
    }
}
